src/DotVVM.Compiler/Initialization/ConfigurationInitializer.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using DotVVM.Compiler.Compilation;
+using DotVVM.Compiler.Exceptions;
+using DotVVM.Compiler.Fakes;
+using DotVVM.Compiler.Resolving;
+using DotVVM.Framework.Compilation.ControlTree;
+using DotVVM.Framework.Configuration;
+using DotVVM.Framework.Security;
+using DotVVM.Framework.Utils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace DotVVM.Compiler.Initialization
+{
+    public class ConfigurationInitializer
+    {
+        internal static DotvvmConfiguration InitDotVVM(Assembly webSiteAssembly, string webSitePath, ViewStaticCompiler viewStaticCompiler, Action<IServiceCollection> additionalServices = null)
+        {
+            var dotvvmStartup = new DotvvmStartupClassResolver().GetDotvvmStartupInstance(webSiteAssembly);
+            var startupClass = new DotvvmServiceConfiguratorResolver().GetServiceConfigureExecutor(webSiteAssembly);
```

Comment:
It's not a startup class. I would rename it to `configureServicesExecutor`. And I think `GetServiceConfigureExecutor` should be `GetConfigureServicesExecutor`.